src/string_array.c

```diff
@@ -98,10 +98,6 @@ rcutils_string_array_cmp(
     lhs, "lhs string array is null", return RCUTILS_RET_INVALID_ARGUMENT);
   RCUTILS_CHECK_FOR_NULL_WITH_MSG(
     rhs, "rhs string array is null", return RCUTILS_RET_INVALID_ARGUMENT);
-  RCUTILS_CHECK_FOR_NULL_WITH_MSG(
-    lhs->data, "lhs->data is null", return RCUTILS_RET_INVALID_ARGUMENT);
-  RCUTILS_CHECK_FOR_NULL_WITH_MSG(
-    rhs->data, "rhs->data is null", return RCUTILS_RET_INVALID_ARGUMENT);
```

Comment:
Without these checks one of the data pointers can be `NULL` which will be accessed below unconditionally:
* https://github.com/ros2/rcutils/pull/246/files#diff-c0cb908ca9bf786997651b24b643070cR118
* https://github.com/ros2/rcutils/pull/246/files#diff-c0cb908ca9bf786997651b24b643070cR120

Reply:
That code is unreachable if `smallest_size` is zero. That's the whole point of moving them into the conditional.

Reply:
I missed that the loop doesn't do any iteration when its zero.